### ISSUE TITLE
`fill_null` expanded on multiple columns

### ISSUE THREAD

**Issue by gab23r** [pola-rs/polars#11634]
### Description

I would like to `fill_null` multiple columns, and I wish I could do it with expression expansion (in one line).

This works:
```python
import polars as pl
df =  pl.DataFrame({'col_1': [0, None], 'col_2': [None, None], 'col_1_fill': [1, 4], 'col_2_fill': [3, 5]})
(
    df.select(
        pl.col('col_1').fill_null(pl.col('col_1_fill')),
        pl.col('col_2').fill_null(pl.col('col_2_fill'))
    )
)
```

**Comment by mcrumiller** [comment]
Just use comprehension:

```python
df.select(
    col(f"col_{x}").fill_null(col(f"col_{x}_fill"))
    for x in range(0, 3)
)
```

**Comment by gab23r** [comment]
Of course, What really surpised me was the error you get when you use the regex version: `ComputeError: an expression is not allowed to have different regexes`. Why different regexes are not allowed ?

For exemple this works:

```
(
    df
    .select(
        pl.col('^col_\d$').fill_null(pl.col('^col_\d$').sort())
    )
)

```

**Comment by gab23r** [comment]
So expansion works if the two expression roots are the same, but I tought that this behavior could the generalized. The two roots should just be of the same size. Just dropping thoughts here, I do not know the complexity behind.

**Comment by cmdlineluser** [comment]
There is a feature request to relax the restriction: https://github.com/pola-rs/polars/issues/8282

Some additional info from the thread: https://github.com/pola-rs/polars/issues/8282#issuecomment-1637290733

> `pl.col('^a-\w*$'`) and `pl.col('^b-\w$')` is not guaranteed to return the same number of columns, and that is why multiple regular expressions aren't allowed.
